### DISCUSSION TITLE
Autodiscover filter passed as an env variable is treated as an array

### DISCUSSION THREAD

**Discussion by catostrophe** [renovatebot/renovate#20668] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.152.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

15.3.5-ee

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

After #16525 and #18531, renovate fails to properly parse the following glob pattern passed from env:
```bash
RENOVATE_AUTODISCOVER_FILTER=my-group/some-prefix-{subrepo1,subrepo2}
```
In the logs, it's shown in the combined config as:
```json
 "autodiscoverFilter": ["my-group/some-prefix-{subrepo1", "subrepo2}"],
```
but should be
```json
 "autodiscoverFilter": ["my-group/some-prefix-{subrepo1,subrepo2}"],
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by catostrophe** [comment]
Linked PRs: #16525 and #18531

**Comment by secustor** [comment, marked as answer]
This is not a bug, but a documented limitation if ENV are used.

> If you use an environment variable or the CLI to set the value for autodiscoverFilter, then commas , within filters are not supported. Commas will be used as delimiter for a new filter.

https://docs.renovatebot.com/self-hosted-configuration/#autodiscoverfilter

Converting to a discussion.

**Reply by catostrophe** [reply]
Thanks.
